addons/mail/models/mail_message.py

```diff
@@ -597,13 +597,29 @@ def _generate_model_record_ids(msg_val, msg_ids):
             ),
         )
 
-    def _validate_access_for_current_persona(self, operation):
-        if not self:
-            return False
-        self.ensure_one()
-        self.sudo(False).check_access_rule(operation)
-        self.sudo(False).check_access_rights(operation)
-        return True
+    @api.model
+    def _get_with_access(self, message_id, operation, **kwargs):
+        """Return the message with the given id if it exists and if the current
+        user can access it for the given operation."""
+        message = self.browse(message_id).exists()
+        if not message:
+            return self.browse()
```

Comment:
```suggestion
            return self.env["mail.message"]
```
Better to see which recordset is returned at a glance.